### ISSUE TITLE
ORM _rec_name compute not work search

### ISSUE THREAD

**Issue by shurshilov** [odoo/odoo#88377]
Impacted versions:
14,15 maybe all

Steps to reproduce:
create new model like this with _rec_name is compute field not store in db
![image](https://user-images.githubusercontent.com/11828278/162687473-2995002e-5a26-423d-af56-48efbfd12f08.png)

Current behavior:
In other model that have many2one relation on this model serach not will work,
always will return all records

Expected behavior:
Do warning in log OR do check on this and get exeption in _rec_name field need store=True

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/11828278/162687919-cd3dc494-9f91-4f07-9e29-9f189c45baf0.png)

Support ticket number submitted via odoo.com/help (optional):

**Comment by gdgellatly** [comment]
You don't need to use store=True although this is an obvious case for it. But if you read the documentation it clearly outlines how to search non stored fields.

**Comment by shurshilov** [comment]
This example does not concern just calculated fields, but concerns using them in _rec_name, it seems to me that it is still more correct to issue a warning if this happens. Because otherwise the standard Odoo functionality is disabled. And custom calculated fields really may not participate in the search and this is NOT a problem. And it's not that we can use the search attribute to implement this. These are two different cases.
As a result, I think if the compute field is used in _rec_name and it has Store=False and there is no search attribute, a warning should be issued to the developer in the log.
This is the same thing that a warning comes out when the _description attribute is missing in the model. And the documentation also has a description of the _description attribute, but despite this warning occurs in the log.

**Comment by gdgellatly** [comment]
Nah, one is an omission of effectively boiler plate, the other is active developer choice of rarely used function in rarely used manner. The developers own unit tests would pick it up.

**Comment by shurshilov** [comment]
> Nah, one is an omission of effectively boiler plate, the other is active developer choice of rarely used function in rarely used manner. The developers own unit tests would pick it up.

It is so rarely used that in half of the modules from the core it is applied:) Do you understand that no one uses the ode out of the box? everyone is working on it. So when referring to such models where the function is calculated and not stored in the database and the search is not done, the search will not work in many2ony)

**Comment by gdgellatly** [comment]
Maybe I'm wrong, honestly I've done for a long time and non stored computed fields for rec_name I can never recall seeing. I struggle to think of a case outside display_name that even uses stored computed fields. 

About the most complex I can think of is product, which cascade searches on barcode, default code, template name and variants and then alters the display with a custom name get depending on context. 

But my point remains, one is an omission, the other is active developer choice and they should test for it, just like the 1000 other things that aren't in logs.
